compiler/parser/python.lalrpop

```diff
@@ -348,8 +348,7 @@ MatchStatement: ast::Stmt = {
             .body
             .last()
             .unwrap()
-            .end_location
-            .unwrap();
+            .end();
```

Comment:
In short term, adding 2 functions will reduce diff from upstream a lot.

```
fn start(&self) -> Location {
    self.start_location
}
fn end(&self) -> Location {
    self.end_location.unwrap()
}
```

Reply:
I'll create a PR for upstreaming this change and look to see if other changes can easily be upstreamed. 

One big question is if `RustPython` wants to rename `Location` to `TextSize`, use a `type Location = TextSize` alias, or change `Location` internally.

Reply:
One more thinking.
We will not change `location` field name, because it is python's naming and we respect them whenever we can.
Not sure about Location. If we can do everything with TextSize, I will name it `Location` because it is a good name as type of `location` fields. But If we cannot avoid to have both `Location` as original and `TextSize`, I will use `TextSize` to avoid confusion.